setup.md

```diff
@@ -1,7 +1,353 @@
 ---
 title: Setup
+teaching: 0
+exercises: 0
+questions:
+- What do I need before starting the SSH tutorial?
+- How can I create and test an SSH key on my computer or a GitHub Codespace?
+- What if I don’t have access to a remote cluster like CERN?
+objectives:
+- Generate and use an SSH key pair.
+- Test an SSH connection inside GitHub Codespaces, locally or through lxplus (at CERN) or (LPC at Fermilab).
+keypoints:
+- SSH provides a secure encrypted connection between your computer and a remote server.
+- SSH keys allow passwordless, secure authentication.
+- Public keys are uploaded to servers; private keys stay on your machine.
 ---
-FIXME
 
+# Overview
+
+Before starting this tutorial, you’ll set up **SSH (Secure Shell)** on your computer or in GitHub Codespaces.
+SSH lets you securely connect to another machine, execute commands remotely, and transfer files.
+
+You will learn how to:
+
+- generate your first SSH key pair
+- test SSH connections (Codespaces or local machine)
+- understand how SSH works
+- optionally connect to lxplus or Fermilab LPC
+
+This setup ensures you’re ready for the hands-on SSH lessons that follow.
+
+---
+
+# 1. Generate an SSH Key
+
+SSH uses **asymmetric cryptography** — a *public key* you can share and a *private key* that must remain secret.
+
+Create your key pair:
+
+```bash
+ssh-keygen -t ed25519 -C "your_email@example.com"
+````
+
+If `ed25519` is not available:
+
+```bash
+ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
+```
+
+Your keys will be created in:
+
+```
+~/.ssh/id_ed25519      # private key (keep safe!)
+~/.ssh/id_ed25519.pub  # public key (share with servers)
+```
+
+---
+
+# 2. How SSH Works
+
+These diagrams illustrate core SSH concepts.
+
+### SSH Connection Handshake
+
+<img src="fig/ssh2_image.png" width="600"/>
+
+### Public/Private Key Authentication
+
+<img src="fig/ssh1_image.png" width="600"/>
+
+---
+
+# 3. Test an SSH Connection (General Form)
```

Comment:
Call this one "Option 1", if the student has access to a remote server like Lxplus, NAF, etc.